sklearn/tree/export.py

```diff
@@ -781,3 +781,144 @@ def export_graphviz(decision_tree, out_file=None, max_depth=None,
     finally:
         if own_file:
             out_file.close()
+
+
+def export_ascii(decision_tree, feature_names=None, class_names=None,
+                 max_depth=10, show_value=False, show_class=False, spacing=3):
+    """Build a text report showing the rules of a decision tree.
+
+    Sample output
+
+    ```
+    |---petal width (cm) <= 0.80
+    |   | (class: setosa)
+    |   |---* value: [ 50.   0.   0.]
+    |   |   | (class: setosa)
+    |---petal width (cm) >  0.80
+    |   | (class: setosa)
+    |   |---petal width (cm) <= 1.75
+    |   |   | (class: versicolor)
+    |   |   |---* value: [  0.  49.   5.]
+    |   |   |   | (class: versicolor)
+    |   |---petal width (cm) >  1.75
+    |   |   | (class: versicolor)
+    |   |   |---* value: [  0.   1.  45.]
+    |   |   |   | (class: virginica)
+    ```
+
+    Parameters
+    ----------
+    decision_tree : decision tree
+        The decision tree to be exported.
+
+    feature_names : list, optional (default=None)
+        A list of length n_features containing the feature names.
+        If None generic names will be used ("feature_0", "feature_1", ...).
+
+    class_names : list of strings, bool or None, optional (default=None)
+        Names of each of the target classes in ascending numerical order.
+        Only relevant for classification, not supported for multi-output.
+        Ignored if show_class is False.
+
+    max_depth : int, optional (default=10)
+        Only the first max_depth levels of the tree are printed.
+
+    show_value : bool, optional (default=False)
+        If True the value of each internal node is printed.
+        Otherwise the value will be reported only for leaves.
```

Comment:
Can we also have an option to disable values, even on leave nodes? In the cases where we want just display the mapping between the features values and output..

Also some link for to the user manual about what `internal node value` in a decision tree corresponds to wouldn't hurt

Reply:
Perhaps `show_value : {'leaves', True, False}`. I'd consider a good default to be: show value at leaves for regression.